### PR TITLE
Fixes #15223: Execute a single directive on the agent

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/15223

### CODE REVIEW

File: share/commands/directive-run [hunk 1]
```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+# @description List directives that can be run on this agent
```
Comment (ncharles) [inline share/commands/directive-run]: Run a specific list of directives

File: share/commands/directive-run [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+major_compare "5.1" "${SERVER_VERSION}"
+if [ $? -eq 255 ]
+then
+  echo "Your server doesn't support this feature, please upgrade to at least 5.1"
+  exit 1
+fi
+if [ ! -f "${BASE_PATH}/rudder-directives.csv" ]
+then
+  echo "Your server didn't generate rudder-directives.csv, maybe you are not up to date"
```
Comment (ncharles) [inline share/commands/directive-run]: or node isn't accepted yet within Rudder ?

File: share/commands/directive-run [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+  canon_uuid=$(echo "${uuid}" | tr '-' '_')
+
+  if [ "${AUDIT}" = "default" ]
+  then
+    [ "${mode}" = "enforce" ] && AUDIT="false"
+    [ "${mode}" = "audit" ] && AUDIT="true"
+  fi
+
+  BS="${BS},set_dry_run_mode_${DRY_RUN}"
```
Comment (ncharles) [inline share/commands/directive-run]: how is DRY_RUN defined ?

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
you have conflict in your PR

File: share/lib/common.sh [hunk 4]
```diff
@@ -120,6 +120,44 @@ modification_time() {
...
+  generation=$(echo "${_generation}"| sed 's/^"\(.*\)"$/\1/')
+  hooks=$(echo "${_hooks}"| sed 's/^"\(.*\)"$/\1/')
+  technique=$(echo "${_technique}"| sed 's/^"\(.*\)"$/\1/')
+  technique_version=$(echo "${_technique_version}"| sed 's/^"\(.*\)"$/\1/')
+  is_system=$(echo "${_is_system}"| sed 's/^"\(.*\)"$/\1/')
+  name=$(echo "${_name}"| sed 's/^"\(.*\)"$/\1/')
+}
+
+# Check that a bootstrap is necessary
+bootstrap_check() {
```
Comment (ncharles) [inline share/lib/common.sh]: this is already in 5.1

File: share/commands/directive-run [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+if [ ${FOUND} -eq 0 ]
+then
+  echo "No such directive ${UUID}, aborting"
+  exit 1
+fi
+
+BS="${BS},rudder_end"
+
+echo rudder agent run ${OPTS} -b "${BS}"
```
Comment (ncharles) [inline share/commands/directive-run]: you don't want the echo there

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**